multiregionccl: gate multi-region database configuration behind a license

otan opened the pull request:
Resolves #59668 

Release justification: low risk changes to new functionality

Release note (enterprise change): Multi-region database creations are
permitted as long as the cluster has a CockroachDB subscription.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/61041)
<!-- Reviewable:end -->

ajstorm reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/61041#-:-MUOVncgA932LNq90SH0:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/ccl/multiregionccl/multiregion_test.go, line 92 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/61041#-MUOVWqt62GWhb0HmGyW:-MUOVWqt62GWhb0HmGyX:b-u7ft9e) ([raw file](https://github.com/cockroachdb/cockroach/blob/521815c8384df9e31220439826f976f0d3dbacfc/pkg/ccl/multiregionccl/multiregion_test.go#L92)):*
> ```Go
> 			// The previous command on t4 does some async cleanup
> 			// which may require retries for the DROP COLUMN to succeed.
> 			{stmt: `ALTER TABLE t4 DROP COLUMN crdb_region`, retry: true},
> ```

Nit: Does this mean that `ALTER DATABASE ADD REGION` will still work?  Not essential, but would be nice if we blocked that as well.


<!-- Sent from Reviewable.io -->

otan commented:
___
*[pkg/ccl/multiregionccl/multiregion_test.go, line 92 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/61041#-MUOVWqt62GWhb0HmGyW:-MUP_C-V1X8gV0cekM6Z:bjhilb5) ([raw file](https://github.com/cockroachdb/cockroach/blob/521815c8384df9e31220439826f976f0d3dbacfc/pkg/ccl/multiregionccl/multiregion_test.go#L92)):*
<details><summary><i>Previously, ajstorm (Adam Storm) wrote…</i></summary><blockquote>

Nit: Does this mean that `ALTER DATABASE ADD REGION` will still work?  Not essential, but would be nice if we blocked that as well.
</blockquote></details>

upcoming PR


<!-- Sent from Reviewable.io -->

otan committed:
multiregionccl: gate multi-region database configuration behind a license

Release justification: low risk changes to new functionality

Release note (enterprise change): Multi-region database creations are
permitted as long as the cluster has a CockroachDB subscription.

otan commented:
bors r=ajstorm
